bundles/org.openhab.binding.shelly/src/main/java/org/openhab/binding/shelly/internal/api/ShellyHttpApi.java

```diff
@@ -489,14 +495,17 @@ private String request(String uri) throws IOException {
                 timeoutsRecovered++;
                 logger.debug("Shelly API timeout recovered");
             } catch (IOException e) {
+                String message = e.getMessage();
                 String type = StringUtils.substringAfterLast(e.getCause().toString(), ".");
-                if (e.getMessage().contains("Timeout") || type.toLowerCase().contains("timeout")
-                        || e.getMessage().contains("Connection reset")) {
-                    throw new IOException(
-                            thingName + ": Shelly API timeout (" + type + "), uri=" + uri);
+                if ((message != null) && (type != null)) {
```

Comment:
Extraneous parens.

Reply:
same, you change would also change the logic